azure-quantum/azure/quantum/target/microsoft/elements/dft/job.py

```diff
@@ -0,0 +1,56 @@
+import json
+from typing import Any, Dict
+from azure.quantum.job.job import Job, DEFAULT_TIMEOUT
+from azure.quantum._client.models import JobDetails
+
+class MicrosoftElementsDftJob(Job):
+    """
+    A dedicated job class for jobs from the microsoft.dft target.
+    """
+
+    def __init__(self, workspace, job_details: JobDetails, **kwargs):
+        super().__init__(workspace, job_details, **kwargs)
+
+    def get_results(self, timeout_secs: float = DEFAULT_TIMEOUT) -> Dict[str, Any]:
+        try:
+            job_results = super().get_results(timeout_secs)
+            return job_results["results"]
+        except RuntimeError:
+            if self.details.status == "Failed":
+                job_blob_properties = self.download_blob_properties(self.details.output_data_uri)
+                if job_blob_properties.size > 0:
+                    job_failure_data = self.download_data(self.details.output_data_uri)
+                    raise DftRuntimeError("An error occurred during DFT-job execution. " \
+                                          "Use error.get_details() for more information.", job_failure_data)
```

Comment:
Maybe move some of this logic to a base class so it can be reused across derived classes. Something like _get_failed_output_or_raise() that returns either what job_failure_data expects or throws

Reply:
Thanks for the suggestion. Moved the logic for getting failure results to the base `Job` class. Defined a "classmethod" `_allow_failure_results()` to allow derived Job-classes to define the failure behavior for themselves.